File: code/Java/Chapter5/5-1/Queue.java
import java.util.LinkedList;

public class Queue<T> {
    private LinkedList<T> data;

    public Queue() {
        data = new LinkedList<T>();
    }

    public boolean isEmpty() {
        return data.isEmpty();
    }

    public int size() {
        return data.size();
    }

    public void enqueue(T elem) {
        data.addLast(elem);
    }

    public T dequeue() {
        return data.removeFirst();
    }

    public T front() {
        return data.getFirst();
    }
}

File: code/Python/Chapter5/5-1/queue.py
class Queue:
    def __init__(self):
        self.__data = []
    
    def is_empty(self):
        return self.__data == []
    
    def __len__(self):
        return len(self.__data)
    
    def clear(self):
        self.__data = []
    
    def enqueue(self, elem):
        self.__data.append(elem)
    
    def dequeue(self):
        return self.__data.pop(0)
    
    def front(self):
        return self.__data[0]